packages/wallet/src/cip30.ts

```diff
@@ -293,18 +294,38 @@ export const createWalletApi = (
     if (shouldProceed) {
       const wallet = await firstValueFrom(wallet$);
       try {
+        const needsForeignSignature = await requiresForeignSignatures(coreTx, wallet);
+
+        // If partialSign is false and the wallet could not sign the entire transaction
+        if (!_partialSign && needsForeignSignature)
```

Comment:
Now that the argument is used, let's rename it
```suggestion
        if (!partialSign && needsForeignSignature)
```